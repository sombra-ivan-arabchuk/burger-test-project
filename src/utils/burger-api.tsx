import nutritionAPI from './api-resourses/nutrition';
import { AxiosResponse } from 'axios';

const fetchInitialItems = (): Promise<AxiosResponse<any>> => {
  return nutritionAPI.get('search/burger', {
    params: { results: '0:20', fields: 'item_id' },
  });
};

export { fetchInitialItems };
