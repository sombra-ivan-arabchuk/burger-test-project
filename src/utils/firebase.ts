import { AxiosResponse } from 'axios';
import burgerAPI from './api-resourses/firebase-burger';
import { BurgerProps } from '../screens/Catalog/Catalog';

const getBurgers = (): Promise<AxiosResponse> => burgerAPI.get('orders.json');

const saveBurger = (burger: BurgerProps): Promise<AxiosResponse> =>
  burgerAPI.post('/orders.json', burger);

const editBurger = (
  id: string | undefined,
  burger: BurgerProps,
): Promise<AxiosResponse> => burgerAPI.put(`/orders/${id}.json/`, burger);

const deleteBurger = (id: string | undefined): Promise<AxiosResponse> =>
  burgerAPI.delete(`orders/${id}.json`);

export { getBurgers, saveBurger, editBurger, deleteBurger };
