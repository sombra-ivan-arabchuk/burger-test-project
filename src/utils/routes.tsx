const routes = {
  home: '/',
  catalog: '/catalog',
};

export default routes;
