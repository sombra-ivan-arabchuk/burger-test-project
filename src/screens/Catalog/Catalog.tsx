import React, { useEffect } from 'react';

import { fetchInitialItems } from '../../utils/burger-api';

const Catalog = (): React.ReactElement => {
  useEffect(() => {
    fetchInitialItems().then(res => {
      // only for now, to see the response
      console.log(res);
    });
  }, []);

  return <div>hello</div>;
};

export default Catalog;
