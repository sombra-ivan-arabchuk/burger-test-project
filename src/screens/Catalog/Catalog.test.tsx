import React from 'react';
import { render, RenderResult } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Catalog from './Catalog';
import * as burgerAPI from '../../utils/burger-api';

import * as firebase from '../../utils/firebase';

export interface SetupProps {
  container: HTMLElement;
  utils: RenderResult;
}

const setup = (): SetupProps => {
  const utils = render(<Catalog />);
  return {
    container: utils.container,
    utils,
  };
};

describe('Catalog', () => {
  it('renders', async () => {
    jest.spyOn(firebase, 'getBurgers').mockImplementation((): any => {
      return Promise.resolve({
        data: {
          LvUY46VP3TPsyNKdHgN: {
            customer: {},
            deliveryMethod: 'fastest',
            ingredients: { bacon: 2, cheese: 2, meat: 2, salad: 2 },
            bacon: 2,
            cheese: 2,
            meat: 2,
            salad: 2,
            price: 5.8,
          },
        },
      });
    });

    jest.spyOn(burgerAPI, 'getIngredientById').mockImplementation((): any => {
      return Promise.resolve([
        {
          LvUY46VP3TPsyNKdHgN: {
            customer: {},
            deliveryMethod: 'fastest',
            ingredients: { bacon: 2, cheese: 2, meat: 2, salad: 2 },
            bacon: 2,
            cheese: 2,
            meat: 2,
            salad: 2,
            price: 5.8,
          },
        },
      ]);
    });
  });
});
