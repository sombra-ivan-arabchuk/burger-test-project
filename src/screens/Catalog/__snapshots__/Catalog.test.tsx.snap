// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog renders 1`] = `
<div
  style="position: fixed; bottom: 5%; right: 10%;"
>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    style="background-color: rgb(240, 142, 73);"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      open builder
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
